_dashboards/dashboards-assistant/text-to-visualization.md

```diff
@@ -0,0 +1,134 @@
+  }
+}
+```
+{% include copy-curl.html %}
+
+### Step 4: Test the agent
+
+You can verify that the agent was created successfully by calling the agent with an example payload:
+
+```json
+POST /_plugins/_ml/agents/<ROOT_AGENT_ID>/_execute
+{
+  "parameters": {
+    "input_question": "find unique visitors and average bytes every 3 hours",
+    "input_instruction": "display with different layers, use independent scale for different layers, display unique visitors with light blue bar chart",
+    "ppl": "source=opensearch_dashboards_sample_data_ecommerce| stats DISTINCT_COUNT(user) as unique_visitors, AVG(taxful_total_price) as avg_bytes by span(order_date, 3h)",
+    "sampleData": """[{\"unique_visitors\":15,\"avg_bytes\":90.98684210526316,\"span(order_date,3h)\":\"2024-04-25 00:00:00\"},{\"unique_visitors\":14,\"avg_bytes\":72.72083333333333,\"span(order_date,3h)\":\"2024-04-25 03:00:00\"}]""",
+    "dataSchema": """[{\"name\":\"unique_visitors\",\"type\":\"integer\"},{\"name\":\"avg_bytes\",\"type\":\"double\"},{\"name\":\"span(order_date,3h)\",\"type\":\"timestamp\"}]"""
+  }
+}
+```
+{% include copy-curl.html %}
+
+## Generating a visualization from text
+
+You can generate a visualization from text by calling the `/api/assistant/text2vega` API endpoint. The  `input_instruction` parameter is optional:
+
+```json
+POST /api/assistant/text2vega
+{
+  "input_instruction": "<input_instruction>",
+  "input_question": "<input_question>",
+  "ppl": "<ppl_query>",
+  "dataSchema": "<data_schema_of_ppl_response>",
+  "sampleData": "<sample_data_of_ppl_response>"
+}
+```
+{% include copy-curl.html %}
+
+The following table describes the Text to Visualization API parameters.
+
+Parameter | Required/Optional | Description 
+:--- | :--- | :---
+`input_question` | Required | The user’s original question used to generate the corresponding Piped Processing Language (PPL) query.
+`ppl` | Required | 	The generated PPL query that retrieves the data required for the visualization.
+`dataSchema` | Required | Describes the structure and types of the data fields in the visualization output, based on the PPL response.
+`sampleData` | Required | Provides sample entries from the data that will populate the visualization.
+`input_instruction` | Optional | Specifies the styling instructions such as colors for the visualization.
+
+## Generating visualizations from text in OpenSearch Dashboards
+
+To generate visualizations from text in OpenSearch Dashboards, use the following steps:
+
+1. On the top menu bar, go to **OpenSearch Dashboards > Visualize**, then select **Create visualization**.
+
+1. In the **New Visualization** dialog, select **Natural language**, as shown in the following image.
+
+    <img src="{{site.url}}{{site.baseurl}}/images/dashboards-assistant/t2viz-start.png" alt="Create a visualization by selecting natural language">
+
+1. From the data sources dropdown list, select a data source, as shown in the following image. 
```

Comment:
Should "data sources" be formatted as a UI element? (**Data sources**)

Reply:
No, this is an unnamed text box